tools/buildHeaders/jsonToSpirv.cpp

```diff
@@ -335,6 +338,19 @@ void jsonToSpirv(const std::string& jsonPath, bool buildingHeaders)
         }
         const auto opcode = inst["opcode"].asUInt();
         const std::string name = inst["opname"].asString();
+        if (firstOpcode) {
+          maxOpcode = opcode;
+          firstOpcode = false;
+        } else {
+          if (maxOpcode > opcode) {
+            std::cerr << "Error: " << name
+                      << " is out of order. It follows the instruction with opcode " << maxOpcode
+                      << std::endl;
+            std::exit(1);
```

Comment:
This may be controversial.  There are also other errors which do not cause immediate termination.
However, those errors are not propagated out to the exit code of the program. I think they should, but can do that separately.